tick/preprocessing/longitudinal_features_product.py

```diff
@@ -0,0 +1,236 @@
+import numpy as np
+
+from tick.base import Base
+from itertools import combinations
+import scipy.sparse as sps
+from scipy.misc import comb
+from sklearn.externals.joblib import Parallel, delayed
+from .build.preprocessing import SparseLongitudinalFeaturesProduct
+
+
+class LongitudinalFeaturesProduct(Base):  # or LongitudinalExposureProduct ?
+    """Transforms longitudinal exposure features to add the corresponding
+    product features. 
+
+    This preprocessor transform an input list of `n_samples` numpy arrays or 
+    csr_matrices of shape `(n_intervals, n_features)` so as to add columns
+    representing the product of combination of two features. It outputs a list
+    of `n_samples` numpy arrays or  csr_matrices of shape `(n_intervals, 
+    n_features + comb(n_features, 2))`.
+    
+    Exposure can take two forms:
+    - short repeated exposures: in that case, each column of the numpy arrays 
+    or csr matrices can contain multiple ones, each one representing an exposure
+    for a particular time bucket.
+    - infinite unique exposures: in that case, each column of the numpy arrays
+    or csr matrices can only contain a single one, corresponding to the starting
+     date of the exposure.
+
+    Parameters
+    ----------
+    exposure_type : `string`, default="infinite"
+        Either "infinite" for infinite unique exposures or "short" for short
+        repeated exposures.
+        
```

Comment:
docstring for n_jobs is missing. 
Maybe this should be called n_threads ? (If I recall well, we use n_threads elsewhere, not n_jobs ?